docs/08-gitcoin-identity-staking-management.md

```diff
@@ -0,0 +1,105 @@
+# Rolling 3-24-Months Identity Staking and re-staking
+
+This describes the contracts and state used for the rolling 3-24 months identity staking and re-staking.
+
+## Data Structures
+
+```solidity
+// uint256 stakeCount;
+uint256 stakeCount;
+
+// mapping(selfStakerAddress => uint[]) selfStakes;
+mapping(address => uint256[]) selfStakes;
+
+// mapping(stakerAddress => mapping(stakeeAddress => uint[])) communityStakes;
+mapping(address => mapping(address => uint256[])) communityStakes;
+
+mapping(uint256 => Stake) stakes;
+
+// mapping(stakerAddress => stakeeAddress[]) addressesStakedOnByUser;
+mapping(address => address[]) addressesStakedOnByUser;
+
+// mapping(stakeeAddress => stakerAddress[]) addressesStakingOnUser;
+mapping(address => address[]) addressesStakingOnUser;
+
+struct Stake {
+  uint256 amount;
+  uint256 unlockTime;
```

Comment:
Let's make this a uint64. that should be sufficient.
And we should also add a lockTime.

Reply:
What is the purpose of the lock time?

Reply:
unlockTime has been updated to uint64